### PR TITLE
feat: update to new DI.jl

### PR DESCRIPTION
**Pull request by avik-pal**

### CODE REVIEW

**Comment by avik-pal** [comment]
The `adjoint` group will fail atm since it won't be able to install SciMLSensitivity

File: Project.toml [hunk 1]
```diff
@@ -34,22 +34,22 @@ SimpleNonlinearSolveTrackerExt = "Tracker"
...
 AllocCheck = "0.1.1"
 Aqua = "0.8"
 ArrayInterface = "7.9"
-CUDA = "5.2"
 ChainRulesCore = "1.23"
 ConcreteStructs = "0.2.3"
 DiffEqBase = "6.149"
 DiffResults = "1.1"
-DifferentiationInterface = "0.5"
+DifferentiationInterface = "0.6"
```
Comment (gdalle) [inline Project.toml:45]:
```suggestion
DifferentiationInterface = "0.6.1"
```

File: test/gpu/cuda_tests.jl [hunk 2]
```diff
@@ -1,73 +1,77 @@
-@testitem "Solving on GPUs" tags=[:cuda] skip=:(using CUDA; !CUDA.functional()) begin
+@testitem "Solving on GPUs" tags=[:cuda] begin
```
Comment (gdalle) [inline test/gpu/cuda_tests.jl:1]: Note that DI is completely untested on GPUs at the moment, so issues may (will) arise

Comment (ChrisRackauckas) [reply test/gpu/cuda_tests.jl:1]: what's the reason for that? Do you not have the buildkite access setup?

Comment (gdalle) [reply test/gpu/cuda_tests.jl:1]: I haven't gotten around to it yet, CPU AD kept me busy enough ^^ but now that the API has stabilized I can look into it

**Comment by ChrisRackauckas** [comment]
Alright, we can merge to start fixing the Enzyme deadlock.

**Pull request merged**